### PR TITLE
Fix secret recreation issue on release-1.0.

### PR DESCRIPTION
**Pull request by myidpt**
Fixes https://github.com/istio/istio/issues/9349.

The function to retrieve service account does not return nil service account when the service account is removed. Existing code relies on this causes recreation of secret. We should rely on the returned error code instead to check if the service account exists.

### CODE REVIEW

File: security/pkg/pki/ca/controller/secret.go [hunk 1]
```diff
@@ -286,8 +286,8 @@ func (sc *SecretController) scrtDeleted(obj interface{}) {
 	}
 
 	saName := scrt.Annotations[ServiceAccountNameAnnotationKey]
-	if sa, _ := sc.core.ServiceAccounts(scrt.GetNamespace()).Get(saName, metav1.GetOptions{}); sa != nil {
```
Comment (gyliu513) [inline security/pkg/pki/ca/controller/secret.go]: Thanks @myidpt for the fix. One question, have you ever test the case of keeping the sa, but just delete the secret, does the secret will be re-created with the fix?

Comment (myidpt) [reply security/pkg/pki/ca/controller/secret.go]: Yes. https://github.com/istio/istio/blob/release-1.0/security/pkg/pki/ca/controller/secret_test.go#L253

**Comment by mandarjog** [comment]
Is this necessary in 1.0 ?

**Comment by lei-tang** [comment]
I think this bug fix should be cherrypicked into Istio 1.0 also.

**Comment by costinm** [comment]
Not critical for 1.0. Please make sure 1.1 is fixed

**Pull request closed without merging**